logging/warning_handler.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logging
+
+import "go.uber.org/zap"
+
+// WarningHandler is a warning handler to forward client-go's warning logs into a zap logger.
+type WarningHandler struct {
+	Logger *zap.SugaredLogger
+}
+
+func (h *WarningHandler) HandleWarningHeader(code int, agent string, message string) {
+	// This condition is copied from K8s' default WarningLogger.
+	if code != 299 || len(message) == 0 {
```

Comment:
almost sure this is a silly question, but do we where 299 comes from / what it means?

Reply:
hah! Found it, it's apparently:

~~~~
   299 Miscellaneous persistent warning
     The warning text MAY include arbitrary information to be presented
     to a human user, or logged. A system receiving this warning MUST
     NOT take any automated action.
~~~~
from https://datatracker.ietf.org/doc/html/rfc2616#section-14.46

Reply:
sort of wish k8s had used a constant or something here but 🤷

Reply:
Can we update the comment with the RFC? I feel the question will come up again